polkadot/node/core/pvf/prepare-worker/src/lib.rs

```diff
@@ -240,6 +294,11 @@ pub fn worker_entrypoint(
 									memory_tracker_stats,
 									#[cfg(target_os = "linux")]
 									max_rss: extract_max_rss_stat(max_rss, worker_pid),
+									peak_alloc: if peak_alloc > 0 {
+										peak_alloc as u64
+									} else {
+										0u64
+									},
```

Comment:
This is because in theory the metric here could go negative.

`start_tracking` resets the counters to zero, so if there's an allocation that was done *before* the `start_tracking`, and if it was deallocated *after* `start_tracking` then, assuming no other allocations are present, the tracked memory usage would go into the negatives.

Reply:
Makes sense. The edge case you describe can happen if the compilation immediately encounters some error and returns before allocating anything. Which makes me wonder if metrics are useful if an error happened. But they can only skew the average, and not the peak, which should be fine.

Overflows would be bad, but I see the counter type is `isize`, so nevermind.